### PR TITLE
Update azure sdk lib

### PR DESCRIPTION
**Pull request by mabhi**
## Change Overview
As per Azure guidelines, few modules from azure library are retiring support from 31 March 2023 . Due to which there is need to replace such modules with azure recommended azure-go-sdk modules.

go-autorest being used in the blockstorage package will be out of support from March 2023. Follow the documentation below to replace it with a new supported package.
https://azure.microsoft.com/en-us/updates/support-for-azure-sdk-libraries-that-do-not-conform-to-our-current-azure-sdk-guidelines-will-be-retired-as-of-31-march-2023/

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [x] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/blockstorage/azure/azuredisk.go [hunk 1]
```diff
@@ -314,42 +306,38 @@ func (s *AdStorage) SnapshotCreate(ctx context.Context, volume blockstorage.Volu
...
-	return snap, nil
+	blockSnapshot.Volume = &volume
+	return blockSnapshot, nil
 }
 
 func (s *AdStorage) SnapshotCreateWaitForCompletion(ctx context.Context, snap *blockstorage.Snapshot) error {
 	err := poll.Wait(ctx, func(ctx context.Context) (bool, error) {
 		snapshot, err := s.SnapshotGet(ctx, snap.ID)
-		if err == nil && snapshot.ProvisioningState == string(azcompute.ProvisioningStateSucceeded) {
+		if err == nil && snapshot.ProvisioningState == string(armstorage.ProvisioningStateSucceeded) {
```
Comment (leuyentran) [inline pkg/blockstorage/azure/azuredisk.go:340]: I questioned whether or not this is the correct package. When we look at the original code, the consts `azcompute.ProvisioningStateSucceeded` is part of a group of 6 const as such to indicate Provisioning states
```go
// ProvisioningState1 enumerates the values for provisioning state 1.
type ProvisioningState1 string

const (
	// ProvisioningState1Creating ...
	ProvisioningState1Creating ProvisioningState1 = "Creating"
	// ProvisioningState1Deleting ...
	ProvisioningState1Deleting ProvisioningState1 = "Deleting"
	// ProvisioningState1Failed ...
	ProvisioningState1Failed ProvisioningState1 = "Failed"
	// ProvisioningState1Migrating ...
	ProvisioningState1Migrating ProvisioningState1 = "Migrating"
	// ProvisioningState1Succeeded ...
	ProvisioningState1Succeeded ProvisioningState1 = "Succeeded"
	// ProvisioningState1Updating ...
	ProvisioningState1Updating ProvisioningState1 = "Updating"
)
```
When we look at the migrated constant `armstorage.ProvisioningStateSucceeded` here, it is part of the following group, which lacks states such as `Deleting` or `Updating`
```go
// ProvisioningState - Gets the status of the storage account at the time the operation was called.
type ProvisioningState string

const (
	ProvisioningStateCreating     ProvisioningState = "Creating"
	ProvisioningStateResolvingDNS ProvisioningState = "ResolvingDNS"
	ProvisioningStateSucceeded    ProvisioningState = "Succeeded"
)

```

Comment (leuyentran) [reply pkg/blockstorage/azure/azuredisk.go:340]: I think a more appropriate choice is [GalleryProvisioningState](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4#GalleryProvisioningState)
```go
// GalleryProvisioningState - The provisioning state, which only appears in the response.
type GalleryProvisioningState string

const (
	GalleryProvisioningStateCreating  GalleryProvisioningState = "Creating"
	GalleryProvisioningStateDeleting  GalleryProvisioningState = "Deleting"
	GalleryProvisioningStateFailed    GalleryProvisioningState = "Failed"
	GalleryProvisioningStateMigrating GalleryProvisioningState = "Migrating"
	GalleryProvisioningStateSucceeded GalleryProvisioningState = "Succeeded"
	GalleryProvisioningStateUpdating  GalleryProvisioningState = "Updating"
)
```

**Review by leuyentran — CHANGES_REQUESTED** [review]
For more context, the review is technically a re-review from https://github.com/kanisterio/kanister/pull/2368/ . My review strategy was mainly maing sure the changes match, albeit that was not entirely successful with `azuredisk.go` due to recent changes in this file. 

The recent changes in this file that need extra review are:

1.  [Implementation of SnapshotCreateWaitForCompletion ](https://github.com/kanisterio/kanister/pull/2494) - review added 
2. [Re-org of code from here ](https://github.com/kanisterio/kanister/pull/2236) - LGTM

File: pkg/blockstorage/helper_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 The Kanister Authors.
```
[the remark this answers is not in the document]
Comment (mabhi) [reply pkg/blockstorage/helper_test.go:1]: Done

File: pkg/kopia/command/storage/secret_utils.go [hunk 3]
```diff
@@ -16,11 +16,10 @@ package storage
 
 import (
 	"context"
-	"time"
-
-	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/kanisterio/kanister/pkg/blockstorage/azure"
```
[the remark this answers is not in the document]
Comment (mabhi) [reply pkg/kopia/command/storage/secret_utils.go:19]: Done

File: pkg/kopia/command/storage/secret_utils.go [hunk 4]
```diff
@@ -16,11 +16,10 @@ package storage
...
 	"github.com/pkg/errors"
 	v1 "k8s.io/api/core/v1"
+	"time"
```
[the remark this answers is not in the document]
Comment (mabhi) [reply pkg/kopia/command/storage/secret_utils.go:22]: Done

**Review by leuyentran — APPROVED** [review]
LGTM, please let Pavan do another pass as well before merging

**Pull request merged**